test/utils/ExternalMulticallSimulator.sol.ts

```diff
@@ -0,0 +1,359 @@
+            ])
+        ).to.be.revertedWith('Sender address not zero');
+      });
+    });
+  });
+
+  describe('tryMulticall via eth_sendTransaction', function () {
+    it('does not revert', async function () {
+      const {
+        roles,
+        externalMulticallSimulator,
+        externalMulticallSimulatorInterface,
+        mockMulticallTarget,
+        mockMulticallTargetInterface,
+      } = await helpers.loadFixture(deploy);
+      await expect(
+        externalMulticallSimulator
+          .connect(roles.randomPerson)
+          .tryMulticall([
+            externalMulticallSimulatorInterface.encodeFunctionData('functionCall', [
+              await mockMulticallTarget.getAddress(),
+              mockMulticallTargetInterface.encodeFunctionData('convertsPositiveArgumentToNegative', [1]),
+            ]),
+          ])
+      ).to.not.be.reverted;
+      expect(await mockMulticallTarget.argumentHistory()).to.deep.equal([]);
+    });
+  });
+
+  // We won't test multicall() and tryMulticall() via eth_call, as they are expected to
+  // behave normally
+
+  // DapiTransmuter is an ExternalMulticallSimulator contract that is given the dAPI name setter role.
+  // OEV searchers can use a DapiTransmuter to simulate multicalls that update dAPIs and make arbitrary calls.
+  describe('DapiTransmuter', function () {
+    it('transmuted dAPI value', async function () {
+      const {
+        roles,
+        externalMulticallSimulator,
+        externalMulticallSimulatorInterface,
+        api3ServerV1,
+        api3ServerV1Interface,
+        ethUsdDapiName,
+        ethUsdDapiProxyWithOev,
+      } = await deployDapiTransmuter();
+      const preTransmutationCalls: Call[] = [];
+      const postTransmutationCalls = [
+        {
+          target: await ethUsdDapiProxyWithOev.getAddress(),
+          data: ethUsdDapiProxyWithOev.interface.encodeFunctionData('read'),
+        },
+      ];
+      const transmutationValue = 123_456;
+      const dapiTransmutationCalldata = await getDapiTransmutationCalldata(
+        preTransmutationCalls,
+        postTransmutationCalls,
+        await api3ServerV1.getAddress(),
+        api3ServerV1Interface,
+        ethUsdDapiName,
+        transmutationValue,
+        externalMulticallSimulatorInterface
+      );
+      const returndata = await externalMulticallSimulator
+        .connect(roles.addressZero)
+        .multicall.staticCall(dapiTransmutationCalldata);
+      // Note that the decoding below is a bit tricky. From the array, the user picks
+      // the item whose returndata they want to read, decode that as a `bytes` type,
+      // then decode the result of that with the types of the returndata of the function
+      // they called.
+      const [transmutedValue] = ethers.AbiCoder.defaultAbiCoder().decode(
+        ['int224', 'uint32'],
+        ethers.AbiCoder.defaultAbiCoder().decode(['bytes'], returndata.at(-1)!)[0]
+      );
+      expect(transmutedValue).to.equal(transmutationValue);
```

Comment:
This might help understand the double-decoding a bit better:

```suggestion
      const readReturndata = returndata.at(-1);
      const [decodedFunctionCallReturndata] = externalMulticallSimulator.interface.decodeFunctionResult(
        'functionCall',
        readReturndata!
      );
      const [transmutedValue] = ethUsdDapiProxyWithOev.interface.decodeFunctionResult(
        'read',
        decodedFunctionCallReturndata
      );
      expect(transmutedValue).to.equal(transmutationValue);
```